src/app/stores/Reducers.js

```diff
@@ -23,35 +24,50 @@ export const workDetail = (state = null, action) => {
   }
 };
 
-export const query = (state = null, action) => (
-  (action.payload) ?
-    action.payload.query :
-    state
-);
+export const searchQuery = (state = null, action) => {
+  if (action.type === Actions.SET_QUERY) {
+    return action.searchQuery;
+  }
 
-export const field = (state = null, action) => (
-  (action.payload) ?
-    action.payload.field :
-    state
-);
+  return state;
+};
 
-export const allowedFilters = (state = null, action) => (
-  (action.payload) ?
-    action.payload.allowedFilters :
-    state
-);
+export const searchField = (state = null, action) => {
+  if (action.type === Actions.SET_FIELD) {
+    return action.searchField;
+  }
+
+  return state;
+};
 
 export const sort = (state = null, action) => (
-  (action.payload) ?
-    action.payload.sort :
+  (action.sort) ?
+    action.sort :
     state
 );
 
-export default combineReducers({
+// export const resetSearch = (state = null, action) => {
+//   if (action.type === Actions.RESET_SEARCH) {
+//     return initialState;
+//   }
+
+//   return state;
+// };
+
+const appReducer = combineReducers({
   searchResults,
-  query,
-  field,
-  allowedFilters,
+  searchQuery,
+  searchField,
   sort,
   workDetail,
 });
+
+export const rootReducer = (state, action) => {
+  if (action.type === Actions.RESET_SEARCH) {
```

Comment:
wouldn't be better to have Reset_Search with it's own reducer? - like the one commented out -

Reply:
I used this answer given to this question by Dan Abramov as an example where the rootReducer resets the state object when a certain action is fired: https://stackoverflow.com/questions/35622588/how-to-reset-the-state-of-a-redux-store. For some reason, the Reducer used separately to return initialState didn't filter down, but I might not have had the correct lifecycle methods running on the components to understand there was a state change.